skorch/tests/test_regressor.py

```diff
@@ -134,7 +134,14 @@ def test_dimension_mismatch_warning(self, net_cls, module_cls, data, recwarn):
         X, y = X[:100], y[:100].flatten()  # make y 1d
         net.fit(X, y)
 
-        w0, w1 = recwarn.list  # one warning for train, one for valid 
+        # protobuf uses deprecated utcnow
+        # TODO: remove once protobuf fixed this
+        w_list = [
+            warning for warning in recwarn.list
+            if "protobuf" not in warning.message.args[0]
+            and "utcnow" not in warning.message.args[0]
+        ]
+        w0, w1 = w_list  # one warning for train, one for valid
```

Comment:
Assuming two warnings by coming out of this filter can be a bit flaky. Can we directly pull out the warnings from training and validation warnings?

Reply:
Yeah, I thought so too. Do you have an elegant solution for that? I was wondering if there was an API for setting filters to `recwarn` but I didn't see anything.

At least, with this implementation, we catch unexpected warnings ;-)

Reply:
After spending too much time fiddling with `pytest.mark.filterwarnings` without success, I rewrote the filters in a similar way as you suggested to make them more robust.